server-common/src/main/java/org/apache/kafka/server/metrics/KafkaMetricsGroup.java

```diff
@@ -0,0 +1,161 @@
+        return KafkaYammerMetrics.defaultRegistry().newGauge(metricName(name, tags), metric);
+    }
+
+    public final <T> Gauge<T> newGauge(final String name, final Gauge<T> metric) {
+        return newGauge(name, metric, Collections.emptyMap());
+    }
+
+    public final Meter newMeter(final String name, final String eventType,
+                                final TimeUnit timeUnit, final Map<String, String> tags) {
+        return KafkaYammerMetrics.defaultRegistry().newMeter(metricName(name, tags), eventType, timeUnit);
+    }
+
+    public final Meter newMeter(final String name, final String eventType,
+                                final TimeUnit timeUnit) {
+        return newMeter(name, eventType, timeUnit, Collections.emptyMap());
+    }
+
+    public final Meter newMeter(final MetricName metricName, final String eventType, final TimeUnit timeUnit) {
+        return KafkaYammerMetrics.defaultRegistry().newMeter(metricName, eventType, timeUnit);
+    }
+
+    public final Histogram newHistogram(final String name, final boolean biased, final Map<String, String> tags) {
+        return KafkaYammerMetrics.defaultRegistry().newHistogram(metricName(name, tags), biased);
+    }
+
+    public final Histogram newHistogram(final String name) {
+        return newHistogram(name, true, Collections.emptyMap());
+    }
+
+    public final Histogram newHistogram(final String name, final boolean biased) {
+        return newHistogram(name, biased, Collections.emptyMap());
+    }
+
+    public final Histogram newHistogram(final String name, final Map<String, String> tags) {
+        return newHistogram(name, true, tags);
+    }
+
+    public final Timer newTimer(final String name, final TimeUnit durationUnit, final TimeUnit rateUnit,
+                                final Map<String, String> tags) {
+        return KafkaYammerMetrics.defaultRegistry().newTimer(metricName(name, tags), durationUnit, rateUnit);
+    }
+
+    public final Timer newTimer(final String name, final TimeUnit durationUnit, final TimeUnit rateUnit) {
+        return newTimer(name, durationUnit, rateUnit, Collections.emptyMap());
+    }
+
+    public final void removeMetric(final String name, final Map<String, String> tags) {
+        KafkaYammerMetrics.defaultRegistry().removeMetric(metricName(name, tags));
+    }
+
+    public final void removeMetric(final String name) {
+        removeMetric(name, Collections.emptyMap());
+    }
+
+    private Optional<String> toMBeanName(final Map<String, String> tags) {
+        final List<Map.Entry<String, String>> filteredTags = tags.entrySet().stream()
+                .filter(entry -> !entry.getValue().equals(""))
+                .collect(Collectors.toList());
+        if (!filteredTags.isEmpty()) {
+            final String tagsString = filteredTags.stream()
+                    .map(entry -> String.format("%s=%s", entry.getKey(), Sanitizer.jmxSanitize(entry.getValue())))
```

Comment:
Let's use string concat (i.e. `+`) instead of `String.format`, the former is more efficient.

Reply:
Ok, done this